app/javascript/contentDisplayPolicy/initHiddenComments.js

```diff
@@ -42,20 +16,65 @@ For further actions, you may consider blocking this person and/or reporting abus
       });
   }
 
+  function showHideCommentsModal(commentId) {
+    const form = document.getElementById('hide-comments-modal__form');
+    form.action = `/comments/${commentId}/hide`;
+
+    window.Forem.showModal({
+      title: 'Confirm hiding the comment',
+      contentSelector: '#hide-comments-modal',
+      overlay: true,
+    }).then(() => {
+      const hideCommentForms = document.querySelectorAll(
+        '.hide-comments-modal__form',
+      );
+
+      hideCommentForms.forEach((form) => {
+        form.addEventListener('submit', handleHideCommentsFormSubmit);
+      });
+    });
+  }
+
   const hideLinks = Array.from(document.getElementsByClassName('hide-comment'));
 
   hideLinks.forEach((link) => {
     const { hideType, commentId } = link.dataset;
-
     if (hideType === 'hide') {
-      link.addEventListener('click', () => {
-        hide(commentId);
+      link.addEventListener('click', (e) => {
```

Comment:
(accessibility / blocking) - because this "Hide" option now triggers a modal, it shouldn't really be a link any more. I think we need to change the HTML to make this a `<button>` element (this is a bit similar to the way we have a "Copy link" button in the post share options dropdown, while the others are links.